Library/brew.rb

```diff
@@ -4,16 +4,17 @@
 
 HOMEBREW_BREW_FILE = ENV['HOMEBREW_BREW_FILE']
 
-if ARGV == %w{--prefix}
-  puts File.dirname(File.dirname(HOMEBREW_BREW_FILE))
-  exit 0
-end
-
 require 'pathname'
 HOMEBREW_LIBRARY_PATH = Pathname.new(__FILE__).realpath.parent.join("Homebrew")
 $:.unshift(HOMEBREW_LIBRARY_PATH.to_s)
+$:.unshift("#{HOMEBREW_LIBRARY_PATH}/test/lib") if ENV["HOMEBREW_TEST_LIB"]
 require 'global'
 
+if ARGV == %w{--prefix}
+  puts HOMEBREW_PREFIX
+  exit 0
+end
```

Comment:
I don't like the idea to move this block. `$(brew --prefix)` is widely used in all kinds of shell scripts. Therefore, the performance is a big concern here. I don't think we should sacrifice this for test.